### ISSUE TITLE
Duplicate local variables aren't renamed

### ISSUE THREAD

**Issue by Gaming32** [QuiltMC/quiltflower#100]
My custom language supports duplicate local variable names (variable hiding) inside scope blocks. This causes the compiler to generate two entries in the `LocalVariableTable` under the same name, and QuiltFlower doesn't rename anything, causing the resulting Java code to be invalid.

Original code:
```groovy
def var a = "hello";
{
    def var a = a.getClass();
    println(a);
}
println(a);
```
CFR output:
```java
import banana.builtin.ModuleBuiltin;

public class ModuleTest {
    private ModuleTest() {
    }

    public static void main(String[] stringArray) {
        String a = "hello";
        Class<?> a2 = a.getClass();
        ModuleBuiltin.println(a2);
        ModuleBuiltin.println((String)a);
    }
}
```
QuiltFlower output:
```java
import banana.builtin.ModuleBuiltin;

public class ModuleTest {
   private ModuleTest() {
   }

   public static void main(String[] args) {
      String a = "hello";
      Class a = a.getClass();
      ModuleBuiltin.println(a);
      ModuleBuiltin.println(a);
   }
}
```
(NOTE: all settings are default)

Original class file: [ModuleTest.zip](https://github.com/QuiltMC/quiltflower/files/8595399/ModuleTest.zip)

**Comment by Gaming32** [comment]
Apparently when the scope block is used inside an if or while, QuiltFlower does rename the variable. My guess as to why is that scope blocks don't exist visibly in the bytecode (except in the `LocalVariableTable`), unless they're in something else, like an if or a while.

**Comment by SuperCoder7979** [comment]
Fixed by [67ee9c5](https://github.com/QuiltMC/quiltflower/commit/67ee9c5243e9dbf40b1cb3607ae210c7b92e719d). Thanks for the report!
